### ISSUE TITLE
off canvas $scope.hide called on resize.body is not default foundation behavior

### ISSUE THREAD

**Issue by reececonrad** [pineconellc/angular-foundation#161]
Just ran across this when testing out an app today.  By default, line 1271 of the tpls version has a $scope.hide function that will toggle the 'move-left' or 'move-right' class off of the off-canvas wrap, hiding the off-canvas navigation.

Then, just below that, is:

win.bind("resize.body", $scope.hide);

So, when the page is resized, the menu hides.  It's not the default behavior of foundation.  The reason I noticed was in mobile safari, when you scroll down from the top of a web page, the url bar slides away and the page technically resizes, thus closing the navigation on scroll.  I can see where this might be a desired effect but maybe it should be mentioned as a change or added as an optional change to the original code

**Comment by janober** [comment]
Did just run into the same issue. Want to have a static side-menu which is only visible on none-mobile and then hides on mobile devices. I add the move-left OR move-right classes myself on resize. Works totally fine and perfect in default foundation but with the angular version it does not because of this issue.

In my eyes it should either be changed to work like the default version or if people really want or need it a parameter should be added. Something like "auto-hide-on-resize".

**Comment by janschaefer** [comment]
Is there any workaround? I have the same problem on mobile devices and a search input in the menu. Whenever you click on the search input, the virtual keyboards opens, resizes the window and the menu hides.

**Comment by horst80** [comment]
@janschaefer, are you sure it's an issue with resize? I had a similar problem, but it was caused by the offCanvasList directive closing the menu on every click. You might have to overwrite it, here's some sample code

```javascript
angular.module("mm.foundation.offcanvas")
        .directive("offCanvasList", [function () {
            return {
                require: "^offCanvasWrap",
                restrict: "C",
                link: function ($scope, element, attrs, offCanvasWrap) {
                    element.find("li").on("click", function (e) {
                        if (angular.element(this).attr("id") === "YOUR_FIELD_ID") {
                            e.stopPropagation();
                        } else {
                            offCanvasWrap.hide();
                        }
                    });
                }
            };
        }]);
```

**Comment by janschaefer** [comment]
Yes, I already have a workaround for the click issue. I am pretty sure it is due to the resize issue. It only happens on mobile devices with virtual keyboards.

**Comment by horst80** [comment]
Have you tried adding a resize listener and detecting if the menu is open? Here's a sample for a controller:

```javascript
controller("OffCanvasController", function ($window) {
    var win = angular.element($window);
    win.bind("resize.body", function (e) {
        var menuOpen = angular.element("div.move-right");
        if (menuOpen.length > 0) {
            e.stopImmediatePropagation();
        }
    });
});
```

Additionally, you could check the window width before and after resize, to detect vertical and horizontal resizes.

**Comment by janschaefer** [comment]
Thank you, that did the job!
